src/test_all.py

```diff
@@ -53,28 +52,26 @@
                 assert type(x['output']) == str, f'the output of example {x} is not a string'
                 assert type(x['explanation']) == str, f'the explanation of example {x} is not a string'
 
-            # if too many samples, sub-sample
             instances = enumerate(data['Instances'])
-            if len(data['Instances']) > 1000:
-                instances = random.sample(list(instances), 500)
 
             # make sure there are no repeated input examples
             for x_idx, x in instances:
                 for y_idx in range(x_idx + 1, len(data['Instances'])):
                     y = data['Instances'][y_idx]
                     if x['input'] == y['input']:
-                        raise Exception(f" * Looks like we have a repeated example here! :-/ \n {x}\n {y}")
-            
+                        raise Exception(f" * Looks like we have a repeated example here! Merge outputs before removing duplicates. :-/ \n {x}\n {y}")
+
             # make sure there are no examples repeated across instances and positive examples
             for x_idx, x in instances:
                 for y_idx in range(len(data['Positive Examples'])):
                     y = data['Positive Examples'][y_idx]
                     if x['input'] == y['input']:
-                        raise Exception(f" * Looks like we have a same example across positive examples and instances! :-/ \n {x}\n {y}")
-            
+                        raise Exception(f" * Looks like we have a same example across positive examples and instances! Merge outputs before removing duplicates. :-/ \n {x}\n {y}")
```

Comment:
Instead of "Merge outputs ..." it should be "Drop the example from the instances".